rbrignall/OU-SUPPS
# new feature: accessibility warnings

Hi Robert,
Thanks for all of your work on `ou-supps`. 

I have this contribution of `accessibility warnings` in this pull request for your review.  It is only implemented for `ouab.cls` at the moment, but if you like the look of it, I would also intend to implement it for the other `.cls` files.

I've put details and a demonstration below. I'd welcome any feedback and questions that you have.
Best
Chris

------------------------

what is this pull request about? why is it helpful/necessary?
-
This work is motivated by accessibility.

I've been converting a lot of our `.tex` files to VLE format. This has included TMAs and exams. There have been several issues that have come up in the conversion that have been caused by the way that authors have typeset their assignments. 

This has created work for me in the conversion; I am hopeful that, by having an `accessibility warnings` option, some authors may enhance or change their typesetting habits.

does this change any existing behaviour?
-
No, this implements new, _optional_ features.

what does this add?
-

1. It adds the command `\uniqueID` to `ouab.cls`
2. It adds the command `\figuredescription` to `ouab,cls`
3. It adds a new option called `accessibility warnings` for `ouab.cls` which can be specified in either of the following, equivalent, ways
```tex
\documentclass[accessibility-warnings]{ouab}
\documentclass[a-w]{ouab}
```

When the `a-w` option is active, then `\jobname.aws` will be created, which will contain `accessibility warnings`; these are outputted to `\jobname.pdf`, and to the log file, `\jobname.log`.

how do I test this?
-
If you compile the following minimal example
```tex
\documentclass[a-w]{ouab}   %<----- NOTE, new option loaded
\begin{document}
an ouab document
\end{document}
```
then you should receive the following pdf (two compilations necessary):

![image](https://user-images.githubusercontent.com/2224480/163331520-f8bf7f9b-c194-46cb-9327-8fb3d6904860.png)

This is _one_ of the accessibility warnings, AW2, that is looked for by `ouab.cls`. 

You should also see the following line in `\jobname.log`
```
 ouab: accessibility warning, AW2: unique ID not specified
```

If you remove the `a-w` option, as in 
```tex
\documentclass{ouab}
\begin{document}
an ouab document
\end{document}
```
then the accessibility warnings are not printed. 

further demonstrations:  ab-accessibility-warnings-demo.tex
-
You will see that I have created a new file ` ab-accessibility-warnings-demo.tex` which contains a full demonstration of the **5** available accessibility warnings: AW1, ..., AW5. 

update to documentation
-
You'll see that I have updated the documentation to account for this. In particular, I have 

- *appended* the `ouab.cls` section
- *appended* Table 1 with a new row for the new key, `unique ID` (currently only `ouab.cls`)
- _added_ a new section 8 on accessibility warnings; a screenshot from Section 8 follows. 

![image](https://user-images.githubusercontent.com/2224480/163332583-c5b8f685-291c-47e9-a7cf-04f1088d6c49.png)

will these accessibility warnings ever be complete?
-
No, we would almost certainly add to these as we become aware of more issues. 

questions for your review
-
I hope that this looks helpful. I thought I'd write the following questions to (hopefully) help with the review:

1. does this feature look helpful?
2. if so, and it is to be accepted, would you like me to implement it for each of `ouexam.cls`, `outn.cls` and `ouicma.cls` before accepting?
